lib/tapioca/compilers/dsl/url_helpers.rb

```diff
@@ -36,7 +36,7 @@ def gather_constants
 
           constants = ObjectSpace.each_object(Module).select do |mod|
             mod = T.cast(mod, T.class_of(Module))
-            next unless mod.name
+            next unless Module.instance_method(:name).bind(mod).call
```

Comment:
What sort of issues we are fixing with this change?

Reply:
If you have a type that redefines the `name` class method, then the value returned from it ends up being useless. For example, if I define a class `Foo`, I would expect `Foo.name == "Foo"` to be true. However, if I defined `Foo` as:
```ruby
class Foo
  def self.name
    nil
  end
end
```
then `Foo.name` would return `nil`, which is not what we want. We want to get the name of the class.

So the solution is to call the `name` instance method of `Module` and call that method on the object that we are working with. So in the above case `Module.instance_method(:name).bind(Foo).call` would correctly return `"Foo"`.

That is what this change is doing.